internal/connection/connection_pool.go

```diff
@@ -103,13 +103,38 @@ func (cp *ConnectionPool) GetByNodeID(driverName, nodeID string) map[string]*Con
 	return result
 }
 
+func (cp *ConnectionPool) getNamespaceByDriverName(driverName string) string {
+	cp.rwlock.RLock()
+	defer cp.rwlock.RUnlock()
+
+	for _, conn := range cp.pool {
+		if conn.DriverName != driverName {
+			continue
+		}
+
+		return conn.Namespace
+	}
+
+	// should be impossible to get here, all Connections have a Namespace
+	return ""
```

Comment:
This may happen if we don't have any connections added to the pool yet.
Let's error out. 
We should have connections added when the request is looped again by reconciler hoepfully.